apis/v1beta1/gateway_types.go

```diff
@@ -571,6 +579,30 @@ type GatewayStatus struct {
 	Listeners []ListenerStatus `json:"listeners,omitempty"`
 }
 
+// GatewayInfrastructure defines infrastructure level attributes about a Gateway instance.
+type GatewayInfrastructure struct {
+	// Labels that should be applied to any resources created in response to this Gateway.
+	//
+	// For implementations creating other Kubernetes objects, this should be the `metadata.labels` field on resources.
+	// For other implementations, this refers to any relevant (implementation specific) "labels" concepts.
+	//
+	// An implementation may chose to add additional implementation-specific labels as they see fit.
+	//
+	// Support: Extended
+	// +kubebuilder:validation:MaxItems=8
```

Comment:
Despite not giving any warning, this line is actually ineffective, we want MaxProperties. Example from TLS Options below (still agree with max of 8 in this case though):

https://github.com/kubernetes-sigs/gateway-api/blob/789326ba892891dcc92d05cae6b9487e63817aca/apis/v1beta1/gateway_types.go#L388